### PR TITLE
boost: update to 1.73.0.

### PR DESCRIPTION
**Pull request by Johnnynator**
There are still quite a few packages that are broken with boost `1.73.0`

### CODE REVIEW

File: srcpkgs/boost/template [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
 maintainer="John <johnz@posteo.net>"
 license="BSL-1.0"
 homepage="http://www.boost.org/"
 distfiles="https://dl.bintray.com/boostorg/release/${version}/source/boost_${version//./_}.tar.bz2"
-checksum=59c9b274bc451cf91a9ba1dd2c7fdcaf5d60b1b3aa83f2c9fa143417cc660722
+checksum=4eb3b8d442b426dc35346235c8733b5ae35ba431690e38c6a8263dce9fcbb402
 
 CXXFLAGS="-std=c++14"
 
-subpackages="libboost_program_options1.72 libboost_math_c99f1.72
```
Comment (sgn) [inline srcpkgs/boost/template:17]: This large diff makes me think about replacing all of 1.73 with:

     ${version%.*}

And hope this list will be stable enough.

Comment (Johnnynator) [reply srcpkgs/boost/template:17]: Sure might be possible, but doing it only in subpackages would not help much. And I don't want to go the libreoffice template route of having a function to generate all sub package functions.

Comment (sgn) [reply srcpkgs/boost/template:17]: Yes, and this is just a `s/1[.]72/1[.]73/g`, hence just a general comment from me.

**Comment by ericonr** [comment]
Can you list broken packages here?

**Comment by Johnnynator** [comment]
Will be revived some when later with a newer boost version.

**Pull request closed without merging**